messaging/src/main/java/org/axonframework/eventhandling/pooled/WorkPackage.java

```diff
@@ -315,16 +315,22 @@ private void processEvents() throws Exception {
                 if (lastStoredToken != lastConsumedToken) {
                     transactionManager.executeInTransaction(() -> storeToken(lastConsumedToken));
                 } else {
-                    transactionManager.executeInTransaction(this::extendClaim);
+                    transactionManager.executeInTransaction(() -> {
+                        extendClaim();
+                        // We update lastClaimExtension here as extendClaim() may be invoked from the outside
+                        lastClaimExtension = clock.instant().toEpochMilli();
```

Comment:
The "lastClaimExtension" can be a valuable field to avoid excessive requests to extend a claim. Instead of moving the field to be used only by the Worker thread, we could also consider making it thread safe (AtomicLong) and use it to ignore requests to extend a claim that are too close to the previous extension request.

Reply:
Smart move. Did this, with some additional changes to ensure the `storeToken` task still occurs frequent enough too.